src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.SslProtocols.cs

```diff
@@ -264,9 +257,25 @@ public void DisabledProtocols_SetSslProtocols_ThrowsException(SslProtocols disab
                 var options = new LoopbackServer.Options { UseSsl = true, SslProtocols = acceptedProtocol };
                 await LoopbackServer.CreateServerAsync(async (server, url) =>
                 {
-                    await TestHelper.WhenAllCompletedOrAnyFailed(
-                        Assert.ThrowsAsync(exceptedServerException, () => server.AcceptConnectionSendResponseAndCloseAsync()),
+                    bool gotExpectedException = false;
+                    Task serverTask = server.AcceptConnectionSendResponseAndCloseAsync();
+                    try
+                    {
+                        await TestHelper.WhenAllCompletedOrAnyFailed(serverTask, 
```

Comment:
I don't think the logic is quite right here.  You want the try/catch and exception check inside the WhenAllCompletedOrAnyFailed call, not around it.  If the serverTask fails first, we'll ignore the result of the other task, and that's not what we want here, I don't think...

I'll put in another shameless plug here for using LoopbackServer.CreateClientAndServerAsync, as it seems like it would clean up the code here.

Similar comment for the other test below.

Reply:
can I wait for tasks separately? I think that would work on this case.

Reply:
Yes, you can just "await task1; await task2;".  

Often you want to wait for either one to fail, and proceed from there, because one task might not complete if the other fails, and then you'll be hung.  I don't think that's the case here, but you can test and confirm.